graders/java/JUnitAutograder.java

```diff
@@ -176,49 +207,54 @@ public synchronized void executionFinished(TestIdentifier test, TestExecutionRes
-            if (autograderTest.message == null)
+            if (autograderTest.message == null) {
                 autograderTest.message = "";
+            }
         }
 
         this.points += autograderTest.points;
     }
 
-    private void saveResults(String signature) {
+    @SuppressWarnings("unchecked")
+    private void saveResults(final String signature) {
 
         double maxPoints = this.classTotals.entrySet().stream().mapToDouble(e -> {
-                if (e.getKey() != null) {
-                    for (TestTag tag : e.getKey().getTags()) {
-                        if (tag.getName().startsWith("maxpoints=")) {
-                            try {
-                                return Double.parseDouble(tag.getName().substring(10));
-                            } catch(NumberFormatException exception) {
-                                JUnitAutograder.this.output = "Could not parse maxpoints tag: " + tag.getName();
-                                JUnitAutograder.this.gradable = false;
-                            }
+            if (e.getKey() != null) {
+                for (TestTag tag : e.getKey().getTags()) {
+                    if (tag.getName().startsWith("maxpoints=")) {
+                        try {
+                            return Double.parseDouble(tag.getName().substring(10));
+                        } catch (NumberFormatException exception) {
+                            JUnitAutograder.this.output = "Could not parse maxpoints tag: " + tag.getName();
+                            JUnitAutograder.this.gradable = false;
                         }
                     }
                 }
-                return e.getValue();
-            }).sum();
+            }
+            return e.getValue();
+        }).sum();
         double testMaxPoints = this.tests.values().stream().mapToDouble(t -> t.maxPoints).sum();
 
         JSONArray resultsTests = new JSONArray();
-        for (AutograderTest test : this.tests.values())
+        for (AutograderTest test : this.tests.values()) {
             resultsTests.add(test.toJson());
+        }
 
         if (maxPoints - testMaxPoints > 0.01) {
             resultsTests.add(new AutograderTest("Incomplete tests", maxPoints - testMaxPoints,
-                                                "The number of points achieved by the autograder tests was not enough to reach \n" +
-                                                "the full amount of tests required for full marks. This is typically caused by \n" +
-                                                "failing early tests, or by an early autograder crash.")
-                             .toJson());
+                    """
+                            The number of points achieved by the autograder tests was not enough to reach\s
+                            the full amount of tests required for full marks. This is typically caused by\s
+                            failing early tests, or by an early autograder crash.""")
```

Comment:
TIL about triple-quote strings in Java! In case anyone else has this question: leading spaces are stripped in this case. I found this helpful to understand how this works: https://www.baeldung.com/java-text-blocks